packages/plugins/graphql-jit/src/index.ts

```diff
@@ -38,35 +38,52 @@ export const useGraphQlJit = (
   const jitCache =
     typeof pluginOptions.cache !== 'undefined' ? pluginOptions.cache : lru<JITCacheEntry>(DEFAULT_MAX, DEFAULT_TTL);
 
-  function getCacheEntry<T>(args: TypedExecutionArgs<T>): JITCacheEntry {
-    let cacheEntry: JITCacheEntry | undefined;
+  function compile(args: ExecutionArgs) {
+    const compilationResult = compileQuery(args.schema, args.document, args.operationName ?? undefined, compilerOptions);
+
+    if (isCompiledQuery(compilationResult)) {
+      return compilationResult;
+    } else {
+      if (pluginOptions?.onError) {
+        pluginOptions.onError(compilationResult);
+      } else {
+        console.error(compilationResult);
+      }
+      return {
+        query: () => compilationResult,
+      };
+    }
+  }
+
+  function getCacheEntry(args: ExecutionArgs): JITCacheEntry {
     const documentSource = documentSourceMap.get(args.document);
 
-    if (documentSource) {
-      cacheEntry = jitCache.get(documentSource);
+    if (!documentSource) {
+      return compile(args);
     }
 
-    if (!cacheEntry) {
-      const compilationResult = compileQuery(args.schema, args.document, args.operationName ?? undefined, compilerOptions);
-
-      if (!isCompiledQuery(compilationResult)) {
-        if (pluginOptions?.onError) {
-          pluginOptions.onError(compilationResult);
-        } else {
-          console.error(compilationResult);
-        }
-        cacheEntry = {
-          query: () => compilationResult,
-        };
-      } else {
-        cacheEntry = compilationResult;
-      }
+    let compiledQuery = jitCache.get(documentSource);
 
-      if (documentSource) {
-        jitCache.set(documentSource, cacheEntry);
-      }
+    if (!compiledQuery) {
+      compiledQuery = compile(args);
+      jitCache.set(documentSource, compiledQuery);
     }
-    return cacheEntry;
+
+    return compiledQuery;
+  }
+
+  function jitExecute(args: ExecutionArgs) {
+    const cacheEntry = getCacheEntry(args);
+
+    return cacheEntry.query(args.rootValue, args.contextValue, args.variableValues);
+  }
```

Comment:
Once I use `makeExecute` to make `args` always an object (to make it v15,v16 compat), it adds an anonymous function on top of the original one which breaks the tests. 
@saihaj